suite/tests/linux/syscall_pwait.c

```diff
@@ -133,9 +140,12 @@ main(int argc, char *argv[])
         /* XXX i#3240: DR currently does not handle the atomicity aspect of this system
          * call. Once it does, please include this in this test or add a new test.
          */
+        sigset_t *test_set_check = &test_set;
         if (pselect(0, NULL, NULL, NULL, NULL, &test_set) == -1) {
             if (errno != EINTR)
                 perror("expected EINTR");
+            if (&test_set != test_set_check)
```

Comment:
This check does not make sense?  The only way this would fail is if the stack slot holding `test_set_check` got clobbered?  How would that happen?  Please explain.

Is this the response to my sigprocmask suggestion?  I was suggesting calling sigprocmask to get the actual current mask, before and after the syscall, to make sure DR is emulating it properly.

Ditto below.

Reply:
Yes it was, that's why I did not put this in originally. Ok I'll add a comment for this.